pkg/trace/agent/agent.go

```diff
@@ -296,17 +296,27 @@ func (a *Agent) convertStats(in pb.ClientStatsPayload, lang, tracerVersion strin
 	in.Lang = lang
 	for i, group := range in.Stats {
 		n := 0
+		updated := false
 		for _, b := range group.Stats {
-			normalizeStatsGroup(&b, lang)
+			if normalizeStatsGroup(&b, lang) {
```

Comment:
not sure if the best way to check if a change was made is to update all the modifier functions.
Maybe we can just run `merge` for each payload.

Reply:
Yes I think that should be fine, it's not like this happens so many times as to warrant an actual performance improvement. Let's start with "nice" code and avoid premature optimisations. Thanks for thinking of that.